mtcpclient/reporting.go

```diff
@@ -38,3 +38,45 @@ func StartBackgroundReporting(numberConnections int, rinterval int) (chan tcpcli
 
 	return connStatusCh, connStatusTracker
 }
+
+func FinalMetricsReport(connectionDescriptions []tcpclient.Connection) string {
+	mr := calculateMetricsReport(connectionDescriptions)
+	return "Time to establish TCP connections min/avg/max/stdDev = " +
+		mr.minToEstablished.String() + "/" +
+		mr.avgToEstablished.String() + "/" +
+		mr.maxToEstablished.String() + "/" +
+		mr.stdDevToEstablished.String()
+}
+
+type metricsReport struct {
+	avgToEstablished    time.Duration
+	minToEstablished    time.Duration
+	maxToEstablished    time.Duration
+	totalToEstablished  time.Duration
+	stdDevToEstablished time.Duration
+}
+
+func calculateMetricsReport(c []tcpclient.Connection) metricsReport {
+	var mr metricsReport
```

Comment:
you can declare this as part of the function definition (func calculateMetricsReport(c []tcpclient.Connection) metricsReport -> func calculateMetricsReport(c []tcpclient.Connection) (mr metricsReport)

Reply:
good to know!